### PR TITLE
Rework Fluid slot behaviour

### PR DESCRIPTION
**Pull request by Natanaelel**
- fluid containers stays in cursor
- return overflow with vanilla `placeItemBackInInventory`
- fix sound not playing when emptying slot completely

## What
When clicking a fluid slot with a fluid container:
- before: Filled container always gets put back in player inventory.
- after: Filled container stays in cursor if possible. If it can't, it gets put back in player inventory.

## Implementation Details
This is not a 1:1 port of 1.12 logic. This is the easiest, and in my opinion the best, implementation of this feature.
The only known difference from 1.12 is when clicking slot containing 5/10B water with a 0.5/1B water fluid cell, 1.12 prefers to fill the slot and empty the cell, whereas gtm and by extension this, prefers to empty the slot and fill the cell.

## Potential Compatibility Issues
This impl assumes that any 1x empty/filled bucket, cell etc. maps to 1x filled/empty bucket, cell etc.

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java [hunk 1]
```diff
@@ -567,23 +566,31 @@ private int tryClickContainer(boolean isShiftKeyDown) {
         if (handler == null) return -1;
         int maxAttempts = isShiftKeyDown ? currentStack.getCount() : 1;
         FluidStack initialFluid = fluidTank.getFluidInTank(tank);
+        FluidStack initialFluidCopy = initialFluid.copy();
```
Comment (krossgg) [inline src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java]: Honestly, we don't even use the `initialFluid` var anymore, so you could just store the fluid type of the initial fluid to get the sound from, eliminating the need for a copy.

File: src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java [hunk 2]
```diff
@@ -567,23 +566,31 @@ private int tryClickContainer(boolean isShiftKeyDown) {
...
+
+                currentStack.shrink(1);
+
+                if (filledResult.isEmpty())
+                    filledResult = remainingStack.copy();
+                else if (ItemStack.isSameItemSameTags(filledResult, remainingStack))
+                    filledResult.grow(1);
+                else {
+                    player.getInventory().placeItemBackInInventory(filledResult);
+                    filledResult = remainingStack.copy();
```
Comment (krossgg) [inline src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java]: Last branch of if/else chain has brackets, so all branches should have brackets for clarity

File: src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java [hunk 3]
```diff
@@ -595,24 +602,36 @@ private int tryClickContainer(boolean isShiftKeyDown) {
...
+
+                currentStack.shrink(1);
+
+                if (drainedResult.isEmpty())
+                    drainedResult = remainingStack.copy();
+                else if (ItemStack.isSameItemSameTags(drainedResult, remainingStack))
+                    drainedResult.grow(1);
+                else {
+                    player.getInventory().placeItemBackInInventory(drainedResult);
+                    drainedResult = remainingStack.copy();
```
Comment (krossgg) [inline src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java]: Last branch of if/else chain has brackets, so all branches should have brackets for clarity

**Review by krossgg — CHANGES_REQUESTED** [review]

**Comment by krossgg** [comment]
Next time, please fill out the template better to provide more information and remember to delete the unused sections (like it says to).

File: src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java (hunk 1 above)
Comment (Natanaelel) [reply src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java]: It gets modified during the draining, which was the reason for the sound bug.

**Review by krossgg — APPROVED** [review]

**Review by YoungOnionMC — APPROVED** [review]

**Pull request merged**